### PR TITLE
add os_server_tag module

### PR DESCRIPTION
**Pull request by n-ik**
##### SUMMARY

Add new module `os_server_tag` to set or remove tag on the openstack server instance

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
module.cloud.openstack.os_server_tag

##### ADDITIONAL INFORMATION
A new module to set and remove tags on openstack server instance.

### CODE REVIEW

**Comment by gtema** [comment]
Generally you do the same what I did in https://github.com/ansible/ansible/pull/48086 and what I did before I landed tagging support in SDK. I have few problems with it:
- min_version of SDK is not mentioned
- when SDK is younger (let's be fair - 90% of cases) there would be no possibility to modify tags

**Comment by n-ik** [comment]
#48086

**Comment by n-ik** [comment]
@gtema 
Thank you for your feedback.
> * min_version of SDK is not mentioned

Good point, I will add this. Do you know since when tags are supported?

> * when SDK is younger (let's be fair - 90% of cases) there would be no possibility to modify tags

You're right, but I think it makes more sense to support the modern way into ansible. 
The other thing is, your way does not support multiple tags.

Do you have any other ideas? I am open for a solution that works for everyone.

**Comment by gtema** [comment]
> > * min_version of SDK is not mentioned
> 
> Good point, I will add this. Do you know since when tags are supported?

0.21.0

> 
> > * when SDK is younger (let's be fair - 90% of cases) there would be no possibility to modify tags
> 
> You're right, but I think it makes more sense to support the modern way into ansible.

You can't even imagine the poorness of the world. In nearly each case I am working with SDK is still 0.12.0 (distro lag, AWX, etc). I am really struggling with that.

> The other thing is, your way does not support multiple tags.

Now it does and also more - it supports managing tags on diverse resources (which support tagging) and also implements using low-level interface when SDK is too old.

**Comment by cloudnull** [comment]
I wonder if this should just be something added to the `os_server` module?

**Comment by n-ik** [comment]
@cloudnull  
I would like to quote @gtema in #48086

* tags are extensions
* tags are on the "separate group" of API description
*  there are server_action, server_group, server_metadata, etc, which are separated, so I followed this pattern not to complicate os_server module
* changing tags does not require (and does not support) setting being part of the "server" request, so it does not really fit os_server.present state (can be of course discussed, but otherwise would require merge of actions, metadata, group etc.)

So basically just KISS and following present pattern.

**Comment by cloudnull** [comment]
@n-ik sorry for the delay. :+1:

**Comment by cloudnull** [comment]
shipit

**Comment by ruizink** [comment]
shipit

**Comment by ruizink** [comment]
shipit

**Comment by ruizink** [comment]
shipit

**Review by cloudnull — APPROVED** [review]
shipit

**Comment by n-ik** [comment]
I fixed the problems from the last test. Can you review again @ruizink @cloudnull. Thank you!!

**Comment by ruizink** [comment]
shipit

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Comment by n-ik** [comment]
Pull request moved to Openstack Gerrit.
